pkg/orchestration/wiring/wiringutil/iam/iam.go

```diff
@@ -28,31 +29,53 @@ const (
 	// See how meta names would clash if Resource Y was named "iamrole" if there was no extra `-`?
 	// See iam_test for the test.
 	namePostfix = "-iamrole"
+
+	// This is just the local reference name
+	dependencyNamePostfix = "iamrole"
 )
 
-func PluginServiceInstance(computeType iam_plugin.ComputeType, resourceName voyager.ResourceName, serviceName voyager.ServiceName, createInstanceProfile bool, dependencyReferences []smith_v1.Reference,
+type ResourceWithIamAccessibleBinding struct {
+	ResourceName               voyager.ResourceName
+	BindableIamAccessibleShape knownshapes.BindableIamAccessible
+	CreatedBindingFromShape    smith_v1.Resource
```

Comment:
If only the name is needed it would be much more efficient to only have the name here. Same with the other similar struct for env vars.